File: app/channels/user_status_channel.rb
class UserStatusChannel < ApplicationCable::Channel
  def subscribed
    stream_from "user_status_#{current_user.id}"

    current_user.update(online: true)
    ActionCable.server.broadcast "user_status_channel", user_id: current_user.id, status: 'online'
  end

  def unsubscribed
    current_user.update(online: false)
    ActionCable.server.broadcast "user_status_channel", user_id: current_user.id, status: "offline"
  end
end

File: app/javascript/channels/user_status_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("UserStatusChannel", {
  connected() {
    console.log("Connected to UserStatusChannel")
  },

  disconnected() {
    console.log("Disconnected from UserStatusChannel")
  },

  received(data) {
    console.log(data)
    const userElement = document.getElementById(`user_status_${data.user_id}`)
    if (userElement) {
      userElement.innerHTML = data.status
      userElement.classList.toggle('online', data.status === 'online')
      userElement.classList.toggle('offline', data.status === 'offline')
    }
  }
});
